### PR TITLE
CFD-248 : Close popup box when "esc" is pressed or click outside popup box

### PR DESCRIPTION
**Pull request by HelenaEksler**
#387

### CODE REVIEW

File: capacity4more/modules/c4m/restful/c4m_restful_quick_post/components/c4m-app/src/controllers/drupalForm.js [hunk 1]
```diff
@@ -154,16 +154,36 @@ angular.module('c4mApp')
...
+        $scope.closePopups();
+      }
+    });
+
+    /**
+     * Make all popups closed.
+     */
+    $scope.closePopups = function() {
+      angular.forEach($scope.popups, function (value, key) {
+        if (name != key) {
```
Comment (amitaibu) [inline capacity4more/modules/c4m/restful/c4m_restful_quick_post/components/c4m-app/src/controllers/drupalForm.js]: add comment.

File: capacity4more/modules/c4m/restful/c4m_restful_quick_post/components/c4m-app/src/controllers/drupalForm.js [hunk 2]
```diff
@@ -154,16 +154,36 @@ angular.module('c4mApp')
...
+          this[key] = 0;
+        }
+      }, $scope.popups);
+      $scope.$digest();
```
Comment (amitaibu) [inline capacity4more/modules/c4m/restful/c4m_restful_quick_post/components/c4m-app/src/controllers/drupalForm.js]: do we need this digest? can't it be inside a $scope.apply?

**Comment by HelenaEksler** [comment]
popup

**Comment by HelenaEksler** [comment]
@amitaibu 
Actually for this situation there is no such a difference between $apply and $digest. So lets use $apply.
PR is ready.

**Comment by zero2one** [comment]
* Clicking outside the popups: popup closed > OK
* Using esc-key : popup closes > OK

But when I want to select a checkbox within the popups, the box closes also without selecting any value:
c4m-taxonomy-widget-click-close

**Comment by HelenaEksler** [comment]
popup1

**Pull request merged**